chia/util/block_cache.py

```diff
@@ -16,18 +19,25 @@ class BlockCache:
 
     _block_records: dict[bytes32, BlockRecord]
     _height_to_hash: dict[uint32, bytes32]
+    mmr_manager: MMRManagerProtocol
 
     def __init__(
         self,
         blocks: dict[bytes32, BlockRecord],
+        mmr_manager: BlockchainMMRManager | None = None,
     ):
         self._block_records = blocks
         self._height_to_hash = {block.height: hh for hh, block in blocks.items()}
+        if mmr_manager is not None:
+            self.mmr_manager = mmr_manager
+        else:
+            self.mmr_manager = BlockchainMMRManager()
 
     def add_block(self, block: BlockRecord) -> None:
         hh = block.header_hash
         self._block_records[hh] = block
         self._height_to_hash[block.height] = hh
+        self.mmr_manager.add_block_to_mmr(block.header_hash, block.prev_hash, block.height)
```

Comment:
one thing we need to be careful with here is the consistency with the blockchain database. Right now, the coin store, the full blocks, peak and height-to-hash all needs to stay in sync, even with a power outage or kernel panic. The MMR manager is similar to the coin store in that it's tied to a specific peak, with a specific chain.

If the database update fails, we may need to roll back this. Or perhaps this only happens after the DB update succeeds.